### DISCUSSION TITLE
Hit an error `position xx does not resolve to a source location` while trying to transform ast to literal code.

### DISCUSSION THREAD

**Discussion by hyf0** [swc-project/swc#2300] [Q&A] [answered]
I write a function like this, some of codes are copied from swc codebase. 
```rust
pub fn ast_to_source(node: &Module) {
  let cm = Lrc::new(SourceMap::new(FilePathMapping::empty()));

  let wr = stdout();
  let mut emitter = Emitter {
      cfg: swc_ecma_codegen::Config { minify: false },
      cm: cm.clone(),
      comments: None,
      wr: Box::new(JsWriter::new(cm.clone(), "\n", wr.lock(), None)),
  };
  emitter.emit_module(node).unwrap();
}
```

While I trying to use it, It panics. I think it's related to `sourcemap`, but I lack the knowledge to solve it.
```rust
let result  = swc_ecma_ast::Module {
        span: DUMMY_SP,
        body,
        shebang: None,
      };
      println!("source");
      ast_to_source(&result)
```

**Comment by kdy1** [comment, marked as answer]
```rust
  let cm = Lrc::new(SourceMap::new(FilePathMapping::empty()));
```

This is wrong. You should use the same instance of `Lrc<SourceMap>` for parsing and codegen.
It is designed to be shared and that's why it's wrapped in `Lrc`, which is `Rc` or `Arc`, depending on the compile option.

**Reply by hyf0** [reply]
The `module` I use to generate is constructing from different `module`s, each module is parsed from a file and each parsing I will generate a new `Lrc<SourceMap>`. Should I also use the same `Lrc<SourceMap>` while parsing?

**Reply by kdy1** [reply]
Yes, you are expected to do so.
`swc_bundler` does the same thing.
